### ISSUE TITLE
development notes

### ISSUE THREAD

**Issue by DoormatIka** [blckzr/Thesis-Project#1]
## repo notes
This repository has the backend and scripts needed to handle training and using the models for inference.

Libraries used are:
- Flask (REST API)
- opencv-python (for live camera feeds, general computer vision tasks, and image manipulation)
- streamlit (for prototype.py, to test models faster with a browser)
- ultralytics (for the YOLOv11n model)

**Read the README.md to see how the code is structured.**

## two-brain architecture
Since YOLO11n will be put in the app background of the phone, we need to include Java or any compiled language that supports the ultralytics models and running in the Android environment.

I haven't set up the environment for that. I need another repo for the android application, since it will use a different language and libraries.

If YOLOv11n isn't natively supported in the language, we can check the ONNX format to see if we can convert it to that (since ONNX is supposed to be universal).

**Comment by DoormatIka** [comment]
okay, so in there's two datasets I need to handle, [deepfakeTIMIT](www.idiap.ch/dataset/deepfaketimit) and [vidTIMIT](https://conradsanderson.id.au/vidtimit/#examples)

deepfakeTIMIT is a modified version of vidTIMIT, notably it *doesn't label which images are deepfaked*. I'm going to assume every video there is deepfaked because most videos there look weird.

so this leads me to the conclusion that vidTIMIT are the original videos.


anyways, vidTIMIT has frames instead of videos like deepfakeTIMIT, so there will be multiple preprocessing steps necessary to make this useable.


for development speed, i advise using the LAV-DF kaggle dataset.

**Comment by DoormatIka** [comment]
**progress report:**

synthesized the dataset. asdlkjlaskl i'm so sleepy.

the dataset's videos got parsed, it had properties (file, number of fake frames in that file, fake periods of seconds, timestamps of each specific word, etc.).

for each video, the video got divided into separate overlapping clips (16 frames per clip). this is to avoid the model getting overwhelmed by the entire video being mostly real but with only a few moments with deepfakes.

anyways, for each clip, i used mediapipe to get the lips of the subject's face to pass it into the model as a tensor, resampled the audio with MFCC for speech recognition, then tacked on the label (fake periods) from the dataset to see if it's real or fake.

all of those are packed into a tuple of three tensors to shove into the mouth of the model.


run the code yourself (scripts/vit/***.py), instructions are at the readme.md, try finding anything wrong with it

**Comment by DoormatIka** [comment]
some thoughts for the paper:

**taglish having no dataset.**

if we're trying to support taglish, we can simply grab a tagalog dataset from one of those multilingual datasets to have something tailored towards the "philippine digital landscape" and an english dataset.


we can test if it works with taglish by showing papers that analyze the taglish dialect, and then showing them that most of taglish are simply conjunctions of english and tagalog words. 

*however*, there's most likely some phonetics about code switching that I don't know about that disrupts this process.

we can explain this as, that the absence of a dedicated Taglish dataset is a constraint, but the combined Tagalog and English training data provides a reasonable approximation given the code-switching nature of the dialect.

this emphasizes how uncertain we are about this, since there aren't many professionally available datasets for taglish, we can rely on approximations.

**Comment by DoormatIka** [comment]
problems to answer for compressed videos focus:

- how do you match the compression algorithms of facebook and viber with ffmpeg?

- ideas:
		- canny edge detection to avoid the compression artifacts/noise from the model?
		-
